### PR TITLE
Dep./CDN updates

### LINKED ISSUE #11: Switch from magvar to geomag
Let's switch over to https://github.com/janrg/wmm-geomagnetic

@janrg noticed an issue https://github.com/dpyeates/magvar/pull/3 and fixed it in his fork

### PR DESCRIPTION
**Pull request by brettz9**
- Enhancement: Use ESM-supporting geomag over magvar; fixes #11
- Enhancement: Add development server and open scripts
- Code docs: List URLs for finding Leaflet, etc. integrity (though now checking this automatically)
- Refactoring: Make separate HTML source file, pointing to local files only
- Refactoring: Use standard ESM in source, moving from browserify to Rollup
- Refactoring: For better performance, move jquery/bootstrap `/vendor` items to CDN (with integrity/crossorigin)
- Refactoring: For better maintenance and modularity, move jquery.easing `/vendor` items to npm
- Refactoring: Make favicon explicit
- npm: Update devDeps.

See what you think. Note that there are a few idiosyncrasies in the now generated HTML file, as cheerio/dom-parser/htmlparser2 that `integrity-matters` is using (to parse the DOM to manipulate it and change `integrity` and update `src`/`href`, etc.):

1. Does not preserve empty quotes like `crossorigin=""`, changing it instead to `crossorigin`
2. Does not preserve inter-attribute whitespace, e.g., when each attribute is on its own line
3. Converts some non-ASCII into entities as is not really necessary.
4. Converts single quotes within double-quotes to entities

However, none of these should have bearing on behavior (the entity conversion makes the output a tiny fraction larger, but should be pretty negligible I think).

I can look into using the same fontawesome CDN as before if you like (just wanted to get this PR in earlier for your review). Or we could make the CDNs all use `unpkg` so that there should only be breakage of unpkg is down.

A few questions:

1. Do you want to add a favicon? If not, I think it'd be nice to at least put a no-op so the server doesn't log distracting HTTP errors for failing to find a favicon. If you did want a favicon, maybe something like a compass? The Greatest Name has limits on how it is supposed to be used, so a compass is the closest I could think of.
2. We could add `cypress` tests (including the code coverage plugin) so there were some automated tests.
3. Since this is not a private repository, I think GitHub technically only allows open source projects. Do you want to settle on a license (and then we can include the license copy and update `package.json` accordingly--even though we are not publishing the repo on npm, it's still good to have the meta-data accurate)?

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    crossorigin="anonymous">
+  <link
+    rel="stylesheet"
+    href="../node_modules/@fortawesome/fontawesome-free/css/fontawesome.css"
+    data-im-algorithms="sha384"
+    crossorigin="anonymous">
+
+  <!-- Keep this path reaching up to `public` so works in both
+       `src` and `public` -->
+  <link href="../public/css/main.css" rel="stylesheet">
```
Comment (wtip) [inline src/index.html]: the `public` directory does not exist once deployed to netlify

**Comment by wtip** [comment]
@brettz9 Sorry for the delay. First let me answer your 3 questions.

1. There already is a favicon that's being used.
2. I don't know if this is really necessary at the moment.
3. I'm open to license suggestions. I'm just not that familiar with the license selection process. A couple concerns I have are that we have code from multiple libraries in the repository, The original site design was based on a MIT licensed template (as is mentioned in the readme) and the photo is Copyright © Bahá'í International Community. So I don't think we can just assign a single license to the entire repository.

If you look below there's a `deploy/netlify — Deploy preview ready!` test that gives you a preview of the site with your pull request. https://deploy-preview-12--qiblih.netlify.app/
It looks like a file path is broken.

**Comment by brettz9** [comment]
No worries on response time! Plan to take a look at the paths issue.

1. Re: favicon, ok, but it is not in the repository. Ok to copy it there?
2. Re: tests, ok.
3. As far as multiple licenses, SPDX expressions as used in `package.json` can be of the form `(MIT AND ...)`. However, as far as the image, technically, it wouldn't be open source if there is copyrighted content. (Some of the content at https://bahai.media/Category:Mansion_of_Bahj%C3%AD may be open if not brought over from https://media.bahai.org/ , but no doubt you want to keep that nice image. Since some of the content is open perhaps you could do `(MIT AND UNLICENSED)`.)

MIT barely adds any restrictions, so, as far as the rest of the project, it depends what your intent is. If you want to compel others to keep any modifications that they may make to the project open, e.g., in case someone were to just copy your site wholesale and improve upon it yet without sharing their improvements with others, and their project started to take away notice of your open version, you could use something like LGPL or MPL. These latter licenses require that modifications to your code would need to be open (though they are not so restrictive as licenses likes the regular GPL to require that all code bundled with the project was also open--I'm not a fan of such latter licenses myself). On the other hand, if you stopped maintaining for some reason, making it MIT would be more likely to ensure that the site was kept going as some might be willing to keep it open if their own code was not obligated to be open.

**Comment by wtip** [comment]
Not sure if I'm misunderstanding what you mean by favicon. This is the file that I'm talking about https://github.com/bahaidev/qiblih-website/blob/master/public/favicon.ico

I'm fine with `MIT AND UNLICENSED` unless you think there's a good reason for something more restrictive.

**Comment by brettz9** [comment]
Re: favicon, whoops, sorry, was looking in the base of the repo, and forgot it was in public. Duh! :)

Re: license, I think `(MIT AND UNLICENSED)` is fine. I think restrictive licenses mostly have their selling point in cases where others may be likely to aggressively try to take ownership, e.g., a company in a competitive or potentially competitive industry (e.g., Linux preventing, by way of its license, a recurrence of the likes of Apple benefiting in such a strong degree from FreeBSD without contributing back and taking away would-be users) or e.g., the Faith's institutions not wanting Covenant-breakers to set up a well-funded alternative site with the Writings that took away attention. It is actually a recognized human right to be able to keep the financial and moral rights for one's creations and use them or share them as one sees fit; I don't think open is better in every situation, but I think it is a healthy default, as it does give more choices to others, which is usually, though not always, a good thing, imho.

**Comment by brettz9** [comment]
Hi @wtip . I think I've gotten it working now (using a `<base>` element in the dev. HTML that `integrity-matters` can now help auto-strip as a convenience). I also explicitly added the favicon.

**Comment by wtip** [comment]
It looks like there's one more issue. If I look at the netlify deploy preview and click on any of the navigation links at the top the smooth scrolling isn't working and I get these errors in the console:
```
scrolling-nav.js:17 Uncaught TypeError: $(...).animate is not a function
    at HTMLAnchorElement.<anonymous> (scrolling-nav.js:17)
    at HTMLAnchorElement.dispatch (jquery-3.5.1.slim.min.js:2)
    at HTMLAnchorElement.v.handle (jquery-3.5.1.slim.min.js:2)
```

**Comment by brettz9** [comment]
I've updated and seems to be working. (I shouldn't have added the slim jquery version.)

**Review by wtip — APPROVED** [review]
Thanks so much for all of your work!

**Pull request merged**